Sources/SwiftSyntax/Syntax.swift

```diff
@@ -202,6 +181,28 @@ public protocol SyntaxProtocol: CustomStringConvertible,
   func childNameForDiagnostics(_ index: SyntaxChildrenIndex) -> String?
 }
 
+// Casting functions to specialized syntax nodes.
+public extension SyntaxProtocol {
+  init?<S: SyntaxProtocol>(_ syntaxNode: S?) {
+    guard let syntaxNode = syntaxNode else {
+      return nil
+    }
+    self.init(syntaxNode)
+  }
+
+  func `is`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> Bool {
+    return self.as(syntaxType) != nil
+  }
+
+  func `as`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S? {
+    return S.init(self._syntaxNode)
```

Comment:
This could be `S(self)` (or `S.init(self)`, right?

Reply:
Ah, sure. I just moved these without really thinking about them. Thanks